lib/hyacinth/adapters/external_identifier_adapter/datacite.rb

```diff
@@ -4,109 +4,136 @@ module Hyacinth
-          raise NotImplementedError
+          rest_api.parse_doi_from_api_response_body(rest_api_response.body)
         end
 
-        # @param id [String]
-        # @param digital_object [DigitalObject]
-        # @param location_uri [String]
-        # @return [Boolean] true if this adapter can handle this type of identifier
-        def update_impl(id, digital_object, location_uri)
-          return update_doi_target_url(id, location_uri) if digital_object.blank?
-
-          # get the metadata from hyacinth
-          hyacinth_metadata = Datacite::Metadata.new digital_object.as_json
-          # prepare the metadata into an acceptable format for EZID
-          datacite_metadata = Datacite::MetadataBuilder.new hyacinth_metadata
-          # ApiSession#modify_identifier returns true if the response from the EZID server indicated
-          # success, else it returns false
-          opts = { datacite: datacite_metadata.datacite_xml, _status: IDENTIFIER_STATUS[:public] }
-          opts[:_target] = location_uri unless location_uri.blank?
-          api_session.modify_identifier(id, opts)
+        # return nil if unsuccessful, else returns the response body
+        def update_doi(doi,
+                       hyacinth_metadata, # Hyacinth::Adapters::ExternalIdentifierAdapter::Datacite::Metadata
+                       target_url, doi_state = nil)
+          datacite_data = Datacite::RestApi::V2::Data.new
+          datacite_data.prefix = DATACITE[:prefix]
+          map_metadata(hyacinth_metadata, datacite_data, target_url)
+          datacite_data.build_properties_update(doi_state)
+          rest_api = Datacite::RestApi::V2::Api.new(DATACITE[:rest_api], DATACITE[:user], DATACITE[:password])
+          rest_api_response = rest_api.put_dois(doi, datacite_data.generate_json_payload)
+          unless rest_api_response.status.eql? 200
+            Rails.logger.error "Did not update a DOI! Response Code: #{rest_api_response.status}." \
+                               "Response Body: #{rest_api_response.body}." \
+                               "Request Body: #{rest_api.most_recent_request_body}"
+            return
+          end
+          rest_api_response.body
         end
 
-        # Following method will make a request to the EZID server to update the target URL associated
-        # with the EZID DOI entry on the server.
-        # (See _target in http://ezid.cdlib.org/doc/apidoc.html#internal-metadata)
-        # To delete the target URL stored in the DOI server, use an empty string as the argument.
-        # Uses the modify identifier API call, returns true if metadata update was successful
-        # if not, returns false
+        # Following only updates the target url. Assumes the other required DataCite properties
+        # are already set. If not sure, use update_doi(doi, hyacinth_metadata, target_url, doi_state = nil)
+        # and supply required metadata
+        # return nil if unsuccessful, else returns the response body
```

Comment:
```suggestion
        # Updates the target url for the given doi, but will fail if all other required DataCite properties
        # have not been previously set for the doi.  If other properties have not been previously set, or if you're
        # unsure whether they've been set, use the update_doi method instead and and supply the required metadata.
        #
        # @param [String] doi A valid doi.
        # @param [String] doi The new target url to set for the given doi.
        # @return [String] response body if successful, or nil upon failure
```
Updated method documentation to rdoc format, and

Reply:
Rdoc used where pertinent.